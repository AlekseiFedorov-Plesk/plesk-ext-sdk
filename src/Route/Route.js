// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement, Component, PropTypes } from '@plesk/ui-library';
import { Route as ReactRoute } from 'react-router-dom';

const stripTags = str => (str || '').replace(/<\w+(\s+("[^"]*"|'[^']*'|[^>])+)?(\/)?>|<\/\w+>/gi, '');

const renderTitle = (title = '') => {
    title = stripTags(title);
    const parts = document.title.split(' - ');

    const documentTitle = document.querySelector('title');
    if (documentTitle) {
        documentTitle.innerHTML = title
            ? `${title} - ${parts[parts.length - 1]}`
            : `${parts[parts.length - 1]}`;
    }

    const heading = document.querySelector('.heading-area');
    if (heading) {
        heading.innerHTML = `<h2><span>${title}</span></h2>`;
    }
};

const getTranslatedTitle = ({ route, locale }) => {
    if (!route) {
        return null;
    }

    const key = `${route.componentPath.replace(/\//g, '.')}.title`;

    return locale.lmsg(key, {}, route.title || null);
};

export default class Route extends Component {
    static propTypes = {
        path: PropTypes.string.isRequired,
        routes: PropTypes.array.isRequired,
    };

    static contextTypes = {
        locale: PropTypes.object,
    };

    componentDidMount() {
        const { path, routes } = this.props;
        const { locale } = this.context;
        const [route] = routes.filter(route => route.path === path);
        const title = getTranslatedTitle({ route, locale });

        renderTitle(title);
    }

    render() {
        return <ReactRoute {...this.props} />;
    }
}
