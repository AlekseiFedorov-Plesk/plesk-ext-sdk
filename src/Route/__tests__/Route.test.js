// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement } from '@plesk/ui-library';
import Route from '../Route';
import { shallow } from 'enzyme';

describe('Route', () => {
    it('renders correctly', () => {
        const routes = [{
            path: '/',
            componentPath: 'Index',
        }];
        const wrapper = shallow(
            <Route
                {...routes[0]}
                routes={routes}
            />,
            {
                context: {
                    locale: {
                        lmsg: jest.fn(() => 'Page title'),
                    },
                },
            }
        );

        expect(wrapper).toMatchSnapshot();
    });
});
