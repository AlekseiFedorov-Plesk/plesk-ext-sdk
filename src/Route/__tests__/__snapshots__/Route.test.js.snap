// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Route
    componentPath="Index"
    path="/"
    routes={
      Array [
        Object {
          "componentPath": "Index",
          "path": "/",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "componentPath": "Index",
      "path": "/",
      "routes": Array [
        Object {
          "componentPath": "Index",
          "path": "/",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "componentPath": "Index",
        "path": "/",
        "routes": Array [
          Object {
            "componentPath": "Index",
            "path": "/",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "locale": Object {
        "lmsg": [MockFunction] {
          "calls": Array [
            Array [
              "Index.title",
              Object {},
              null,
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "Page title",
            },
          ],
        },
      },
    },
    "hydrateIn": undefined,
  },
}
`;
